esp/main/src/TemperatureSensor.cpp

```diff
@@ -0,0 +1,127 @@
+//
+// Created by lukaszk on 09.07.2020.
+//
+
+#include "../include/TemperatureSensor.h"
+#include "Free.cpp"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include <iostream>
+#include <string>
+
+TemperatureSensor::TemperatureSensor() {
+    vTaskDelay(2000.0 / portTICK_PERIOD_MS);
+    oneWireInterface = owb_rmt_initialize(&rmtDriverInfo,
+                                          TEMPERATURE_PIN,
+                                          RMT_CHANNEL_1,
+                                          RMT_CHANNEL_0);
+    owb_use_crc(oneWireInterface, true); // enable CRC check for ROM code
+    owb_search_first(oneWireInterface, &searchState, &found);
+}
+
+void TemperatureSensor::FindDevices() {
+    std::cout << "Found devices:" << std::endl;
+    while (found) {
+        std::string device = TemperatureSensor::CreateStringFromRom(
+            searchState.rom_code);
+        std::cout << NoDevices << " : " << device << std::endl;
+        if (searchState.rom_code == knownDevices[0] ||
+            searchState.rom_code == knownDevices[1]) {
+            //            device_rom_codes[NoDevices] = searchState.rom_code;
+            NoDevices++;
+        }
+        owb_search_next(oneWireInterface, &searchState, &found);
+        totalDevicesNo++;
+    }
+    if (NoDevices != totalDevicesNo) {
+        std::cout << "Unwanted devices detected!!! Total NoDevices: "
+                  << totalDevicesNo << std::endl;
+    }
+    std::cout << "Found " << NoDevices << "device"
+              << (NoDevices == 1 ? "" : "s") << std::endl;
+    if (NoDevices != 2)
+        esp_restart();
+}
+
+void TemperatureSensor::InitializeDevices() {
+    for (int i = 0; i < NoDevices; i++) {
+        DS18B20_Info* ds18b20_info = ds18b20_malloc(); // heap allocation
```

Comment:
Oh thats... not even usage of new... Overload new operator with this custom malloc and make usage of smart pointer there.